src/main/webapp/js/components/interface/3dCanvas/ColorController.js

```diff
@@ -125,7 +125,7 @@ define(['jquery'], function () {
          */
         removeColorFunction: function (instances) {
             while (this.litUpInstances.length > 0) {
-                this.clearColorFunctions(instances[0]);
+                this.clearColorFunctions(instances.pop(0));
```

Comment:
@mattearnshaw what's pop?

Reply:
mutates `instances` by removing its last element and returns that element (the argument here is actually a mistake, `pop` takes no arguments, but we don't care about order so that's fine). but I realise I introduced a mistake in the previous line by the assumption that `instances = this.litUpInstances`, i.e. that this function is always removing **all** the lit up instances. I will revert [this](https://github.com/openworm/org.geppetto.frontend/commit/45c9131fbad206d71caadcd6ff4af7530fe65688), but then the `pop` is required because otherwise the loop is infinite: `clearColorFunctions` does not mutate `instances`.